yt/chyt/server/subquery.cpp

```diff
@@ -426,7 +427,7 @@ class TInputFetcher
 
             for (auto& dataSlice : InputDataSlices_[tableIndex]) {
                 YT_VERIFY(!dataSlice->IsLegacy);
-                dataSlice->SetInputStreamIndex(operandIndex);
+                dataSlice->SetInputStreamIndex(tableIndex);
```

Comment:
And I don't understand that one either

Reply:
In order for sorted pool to work properly, we need to attribute slices to correct input streams: the pool uses these streams to decide whether to use chunk slice fetcher (used for unversioned streams only) and in the end it validates that output slices from a single stream are actually sorted. I am pretty sure these indexes are also used while building jobs, otherwise it would be impossible to have multiple potentially intersecting sorted tables as input.

Previously, we used the operand index as input stream index, so all concatenated tables seemed like one big table to the pool. This has no chances of working, since this "one big table" is not actually sorted.